app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    category = benchmark_run.result.keys[0]
+
+    previous_benchmark_runs = BenchmarkRun.where('created_at < ?', benchmark_run.created_at).where(initiator_type: Commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
That cleared a bit. Another query that I provided was to fetch only the required benchmark runs which are created before our `commit` of interest. This ensures that all the commits which are created after our `commit` and before our benchmark regression job is finished are handled and not taken into account. So for a server running commit 2 first instead of commit 1 will have a benchmark regression test for commit 2 first and then commit 1. Is there a problem?